### PR TITLE
Add basic permalink component

### PR DESCRIPTION
**Pull request by edellucien**

### CODE REVIEW

File: src/Layer.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ export default class Layer extends Observable {
   getVisibleChildren() {
     for (let i = 0; i < this.children.length; i += 1) {
       if (this.children[i].getVisible()) {
-        return true;
+        return this.children[i];
```
Comment (oterral) [inline src/Layer.js]: it should return an array with all the visible children shouldn't it?

File: styleguide.config.js [hunk 2]
```diff
@@ -3,6 +3,9 @@ const { version } = require('./package.json');
 
 module.exports = {
   title: `React-spatial ${version}`,
+  template: {
+    favicon: 'src/images/favicon.png',
```
Comment (oterral) [inline styleguide.config.js]: I don't know if it necessary, can you use : path.join(__dirname, ' ?

File: src/Layer.js (hunk 1 above)
Comment (edellucien) [reply src/Layer.js]: Now it does :)

File: styleguide.config.js (hunk 2 above)
Comment (edellucien) [reply styleguide.config.js]: I tried with path.join but didn't work atm.

File: src/Layer.js (hunk 1 above)
Comment (edellucien) [reply src/Layer.js]: I mean, now ;)

File: src/components/Permalink/Permalink.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import qs from 'query-string';
+
+const propTypes = {
+  /**
+   * params to be written in url.
```
Comment (oterral) [inline src/components/Permalink/Permalink.js]:
```suggestion
   * Params to be written in url.
```

File: src/components/Permalink/Permalink.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+   */
+  params: PropTypes.object.isRequired,
+
+  /**
+   * either 'react-router' history object:
```
Comment (oterral) [inline src/components/Permalink/Permalink.js]:
```suggestion
   * Either 'react-router' history object:
```

**Comment by oterral** [comment]
Nice one!!

**Pull request merged**